src/nvim/screen.c

```diff
@@ -2185,16 +2185,24 @@ static int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow,
       }
     }
 
+    line = ml_get_buf(buf, lnum, false);
     has_decor = decor_redraw_line(wp->w_buffer, lnum-1,
                                   &decor_state);
 
+    int win_lines = plines_win(wp, lnum, true);
+    int leftoffset = wp->w_leftcol + wp->w_skipcol;
+    int rightoffset = leftoffset + win_lines * (wp->w_width - win_col_off(wp));
+    rightoffset = win_linetabsize(wp, line, rightoffset);
```

Comment:
this function takes a byte count and returns a screen width, so this cannot be right.

Reply:
In the tests I did, it seems that is returns a correct value everytime, even with `wrap` on.

Reply:
Though with `linebreak` this seems to be a little borked

Reply:
Bascically I want to compute the amount of text of this line that I am going to see on the screen

Reply:
Even with non-ASCII characters present? What about if the line contains `börk börk börk` or `bière` (or, more likely, `β`)?

Reply:
don't forget unprintable chars, hard tabstops, conceals, etc etc.